lock_api/src/lib.rs

```diff
@@ -83,10 +83,11 @@
 //! - `nightly`: Enables nightly-only features. At the moment the only such
 //!   feature is `const fn` constructors for lock types.
 
-#![no_std]
+#![cfg_attr(not(feature = "i-am-libstd"), no_std)]
 #![warn(missing_docs)]
 #![cfg_attr(feature = "nightly", feature(const_fn))]
 
+#[cfg(not(feature = "i-am-libstd"))]
 #[macro_use]
 extern crate scopeguard;
```

Comment:
I would prefer if you just included a minimal implementation of scopeguard directly in the crate as a module. Like [this](https://gist.github.com/Amanieu/a46426c7d5f2e6e76f8f8f9935830469). You would then be able to entirely drop the scopeguard dependency.

Reply:
Ok. Will look at that.